migrations/1609549880732-CreateMembersTable.ts

```diff
@@ -0,0 +1,61 @@
+import {MigrationInterface, QueryRunner, Table} from "typeorm";
+
+export class CreateMembersTable1609498281554 implements MigrationInterface {
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.createDatabase('app', true);
+        await queryRunner.createTable(new Table({
+            name: 'members',
+            columns: [
+                {
+                    name: 'id',
+                    type: 'int',
+                    isPrimary: true,
+                    generationStrategy: 'increment',
+                    isGenerated: true
+                },
+                {
+                    name: 'firstName',
+                    type: 'varchar',
+                    length: '80'
+								},
+								{
+                    name: 'lastName',
+                    type: 'varchar',
+                    length: '80'
+								},
+								{
+									name: 'shortBio',
+									type: 'text'
+								},
+                {
+                    name: 'imageUrl',
+                    type: 'varchar',
+                    length: '255',
+                    isNullable: true
+                },
+                {
+                    name: 'events',
+                    type: 'text',
+                    isNullable: true
+                },
+                {
+                    name: 'createdAt',
+                    type: 'timestamp',
+                    default: 'CURRENT_TIMESTAMP'
+                },
+                {
+                    name: 'updatedAt',
+                    type: 'timestamp',
+                    default: 'CURRENT_TIMESTAMP',
+                    onUpdate: 'CURRENT_TIMESTAMP'
+                }
+            ]
+        }));
+    }
```

Comment:
Okay, this is as great learning opportunity for you 😄 

We have a many-to-many relationship. We have two tables that we want to connect via this relationship - events and members. To create a many-to-many relationship, we need to create a third table. This third table is called a "linking table". You can see an example in TypeORM's explanation in their documentation: https://typeorm.io/#/many-to-many-relations/what-are-many-to-many-relations. In this method, you also don't need to track the events in this table. The linking table tracks them.

However, their default naming convention for the linking table is funky. So back in the MemberModel class, you can update your `@JoinTable()` annotation to include the linking table name: https://github.com/typeorm/typeorm/blob/master/docs/relations.md#jointable-options

To summarize, you'll need to add creation of the linking table in this migration (or create another migration for it. Your choice). You'll also want to remove the `events` field from the `members` table. Following that, you'll want to update the `@JoinTable()` annotation in MemberModel to reference your new linking table's name.

You can also see this article to help understand many-to-many relationships: https://dzone.com/articles/how-to-handle-a-many-to-many-relationship-in-datab

Reply:
i read the docs and article. I will need some help with this one. Big picture, I understand what many-to-many is, and why another join table is needed. But would like to see a more detailed example than the docs' example.

Reply:
@simonhlee97 I hope [this link](https://github.com/typeorm/typeorm/tree/master/sample/sample4-many-to-many/entity) helps😀  It contains many examples about how TypeORM does the many-to-many relationships between the `Post` table and other tables.

> To summarize, you'll need to add creation of the linking table in this migration (or create another migration for it. Your choice). You'll also want to remove the events field from the members table. Following that, you'll want to update the @JoinTable() annotation in MemberModel to reference your new linking table's name.

I want to add one more thing here. I think we don't need a cascade between the `events` and the `members`. @TheBeege please let me know if I'm wrong.